kratos/utilities/assign_unique_model_part_collection_tag_utility.cpp

```diff
@@ -101,13 +101,17 @@ void AssignUniqueModelPartCollectionTagUtility::ComputeTags(
         if (r_value.size() > 1) combinations[r_value] = 0;
     }
 
-    /* Combinations */
-    for(auto& combination : combinations) {
-        const std::set<IndexType>& r_key_set = combination.first;
-        for(IndexType it : r_key_set)
-            rCollections[tag].push_back(rCollections[it][0]);
-        combinations[r_key_set] = tag;
-        ++tag;
+    if (DataCommunicator::GetDefault().IsDistributed()) {
```

Comment:
please use the data comm of the ModelPart, the default should only be used in tests

Reply:
Ok!